src/Rule/Nested.php

```diff
@@ -81,30 +79,42 @@ protected function validateValue($value, ValidationContext $context = null): Res
                 'Value should be an array or an object. %s given.',
                 gettype($value)
             ));
+
             return $result;
         }
+
         $value = (array) $value;
 
         foreach ($this->rules as $valuePath => $rules) {
-            $rulesSet = is_array($rules) ? $rules : [$rules];
             if ($this->errorWhenPropertyPathIsNotFound && !ArrayHelper::pathExists($value, $valuePath)) {
-                $result->addError(
-                    $this->formatMessage(
-                        $this->propertyPathIsNotFoundMessage,
-                        [
-                            'path' => $valuePath,
-                        ]
-                    )
-                );
+                $message = $this->formatMessage($this->propertyPathIsNotFoundMessage, ['path' => $valuePath]);
+                $result->addError($message);
+
                 continue;
             }
+
+            $ruleSet = is_array($rules) ? $rules : [$rules];
+            $aggregatedRule = new Rules($ruleSet);
             $validatedValue = ArrayHelper::getValueByPath($value, $valuePath);
-            $aggregateRule = new Rules($rulesSet);
-            $itemResult = $aggregateRule->validate($validatedValue);
-            if ($itemResult->isValid() === false) {
-                foreach ($itemResult->getErrors() as $error) {
-                    $result->addError($error);
+            $itemResult = $aggregatedRule->validate($validatedValue);
+            if ($itemResult->isValid()) {
+                continue;
+            }
+
+            foreach ($itemResult->getErrorObjects() as $error) {
+                if (is_array($valuePath)) {
```

Comment:
`$valuePath` always is integer or string. This condition always is false.

Reply:
:heavy_check_mark: